### PR TITLE
Basic Referrals UI

### PR DESCRIPTION
**Pull request by stephencelis**
Some basic UI, more tests, and another edge case.

### CODE REVIEW

File: Sources/PointFree/Subscribe.swift [hunk 1]
```diff
@@ -215,6 +215,7 @@ private func validateReferrer(
     }
 
     let isSubscribeDataValidForReferral = subscribeData.pricing.lane == .personal
+      && user.referrerId == nil
```
Comment (stephencelis) [inline Sources/PointFree/Subscribe.swift]: Missed this edge case.

File: Sources/PointFree/SubscribeConfirmation.swift [hunk 2]
```diff
@@ -78,6 +78,19 @@ private func validateReferralCode(
       )
     }
 
+    guard currentUser?.referrerId == nil else {
```
Comment (stephencelis) [inline Sources/PointFree/SubscribeConfirmation.swift]: And this one.

File: Sources/Views/SubscriptionConfirmation.swift [hunk 3]
```diff
@@ -767,17 +767,25 @@ function updateSeats() {
...
+        + (monthly ? "" : " times <strong>12 months</strong>")
+        + "."
+    )
+  } else {
+    document.getElementById("pricing-preview").innerHTML = (
+      monthly
+        ? "You and your referrer will receive an <strong>$18 credit</strong> when you subscribe. It will apply to future invoices."
+        : "You and your referrer will <strong>save $18</strong> when you subscribe. You will be charged $150 today and $168 on renewal."
+    )
+  }
```
Comment (stephencelis) [inline Sources/Views/SubscriptionConfirmation.swift]: More JS in order to dynamically display different text :/ I'd like to clean some of this up but maybe we decide on how we want this to look first.

File: Sources/Views/SubscriptionConfirmation.swift [hunk 4]
```diff
@@ -767,17 +767,25 @@ function updateSeats() {
...
+        + " per month</strong>"
+        + (seats > 1 ? " times <strong>" + seats + " seats</strong>" : "")
+        + (monthly ? "" : " times <strong>12 months</strong>")
+        + "."
+    )
+  } else {
+    document.getElementById("pricing-preview").innerHTML = (
+      monthly
+        ? "You and your referrer will receive an <strong>$18 credit</strong> when you subscribe. It will apply to future invoices."
+        : "You and your referrer will <strong>save $18</strong> when you subscribe. You will be charged $150 today and $168 on renewal."
```
Comment (mbrandonw) [inline Sources/Views/SubscriptionConfirmation.swift]: is it possible to do this without hard coding the values?
